inspector/src/components/graph/canvasGraphService.ts

```diff
@@ -170,15 +171,10 @@ export class CanvasGraphService {
         // Get global min max of time axis (across all datasets).
         this._globalTimeMinMax.min = Infinity;
         this._globalTimeMinMax.max = 0;
-
-        // First we must get the end positions of each dataset.
-        this.datasets.forEach((dataset: IPerfDataset) => {
-            // skip hidden and empty datasets!
-            if (dataset.data.length === 0 || !!dataset.hidden) {
-                return;
-            }
-
-            const pos = this._positions.get(dataset.id) ?? dataset.data.length - 1;
+            
+        // First we must get the end positions of our view port.
+        if (this._getNumberOfSlices() > 0) {

```

Comment:
could save this._getNumberOfSlices() ??? and reuse

Reply:
Yeah looking at the code I could probably just not even save, but just check once and return early rather than every few lines, because theres nothing to do really if numberOfSlices === 0.

Reply:
Okay yeah i still had to save because of some other uses like end - this._getNumberOfSlices and such, but I also  exit early.